src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,54 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+
+    private static final int MAX_SIZE = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_SIZE];
+        values = (V[]) new Object[MAX_SIZE];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        if (!invalidSize()) {
+            throw new IllegalStateException("Max size is 10");
+        }
+
+        for (int i = 0; i < size; i++) {
+            if ((key == null && keys[i] == null) || (key != null && key.equals(keys[i]))) {
+                values[i] = value;
+                return;
+            }
+        }
+
+        if (size < MAX_SIZE) {
+            keys[size] = key;
+            values[size] = value;
+            ++size;
+        }
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < size; ++i) {
+            if ((key == null && keys[i] == null) || (key != null && key.equals(keys[i]))) {
+                return values[i];
+            }
+        }
         return null;
     }
 
     @Override
     public int size() {
-        return -1;
+        return size;
+    }
+
+    private boolean invalidSize() {
```

Comment:
method names should be verbs, so that we understand that they do something
```suggestion
    private boolean checkSize() {
```
or
```suggestion
    private boolean isSizeValid() {
```